src/synapse/azext_synapse/custom.py

```diff
@@ -0,0 +1,171 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger  # pylint: disable=unused-import
+from knack.util import CLIError
+
+from azext_synapse.vendored_sdks.azure_synapse.models import ExtendedLivyBatchRequest, LivyStatementRequestBody, \
+    ExtendedLivySessionRequest
+
+from azext_synapse.vendored_sdks.azure_mgmt_synapse.models import Workspace, WorkspacePatchInfo, ManagedIdentity, \
+    DataLakeStorageAccountDetails, \
+    BigDataPoolResourceInfo, AutoScaleProperties, AutoPauseProperties, LibraryRequirements, NodeSize, NodeSizeFamily, \
+    SqlPool, Sku
+
+
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements, unused-argument
+
+
+def list_spark_batch_jobs(cmd, client, workspace_name, spark_pool_name, from_index=None, size=None, detailed=True):
+    return client.list(workspace_name, spark_pool_name, from_index, size, detailed)
+
+
+def create_spark_batch_job(cmd, client, workspace_name, spark_pool_name, job_name, args, driver_memory, driver_cores,
+                           executor_memory, executor_cores, num_executors, job_type="SCALA",
+                           file=None, class_name=None, jars=None, files=None,
+                           archives=None, conf=None, artifact_id=None,
+                           tags=None, detailed=True):
+    dotnet_file = "local:///usr/hdp/current/spark2-client/jars/microsoft-spark.jar"
+    dotnet_class = "org.apache.spark.deploy.dotnet.DotnetRunner"
+
+    if job_type.upper() != "DOTNET" and (not file or not class_name):
+        raise CLIError('Scala and python spark batch job must provide value for parameter file and class_name.'

```

Comment:
>python [](start = 34, length = 6)

Python

Reply:
fixed

---
In reply to: [386422281](https://github.com/Azure/azure-cli-extensions/pull/1339#discussion_r386422281) [](ancestors = 386422281)